provisioner/src/lib.rs

```diff
@@ -531,17 +579,23 @@ impl Provisioner for MyProvisioner {
                 // Skip updating subscriptions for admin users, and only update subscription if the
                 // rds instance is new.
                 if claim.tier != AccountTier::Admin && created_new_instance {
+                    let instance_name = format!("{}-{}", &request.project_name, engine);
+
                     // If the subscription update fails, e.g. due to a JWT expiring or the subject's
```

Comment:
Not part of your code, but this seems extremely risky from how it's defined. If the JWT expires (I don't have the full spec in mind, but I feel like this could happen for whatever reason), we're deleting a database with user's data. If we can't connect to the auth service, we're going to delete the database as well; if the subscription has expired (they forgot to pay, oops), we're deleting all of your data.

Reply:
The RDS provisioning is only called for new instances at present, if it has already been provisioned it is returned from resource-recorder. In the past it was called each time, however, and the password was cycled. Even so, to guard against deleting databases with data if this code changes or we start calling it after creation, there is a guard as well. When the rds instance is provisioned, it also returns a bool, the bool indicates whether a new instance was created. The instance is only deleted if it's created in the same call, to avoid creating instances that do not have an associated subscription item.

While I don't think it's high risk at the moment, it is high consequence, so I can add more comments to guard against code changes leading to accidental deletion in the future. Or more guards if you think it's necessary.

Reply:
Ah right, I've missed the boolean part. Reviews late in the day aren't the best ideas. Thanks for the detailed explanation, all clear now :pray:

Reply:
No problem, I appreciate the sanity check!